src/Symfony/Bridge/PhpUnit/DnsMock.php

```diff
@@ -0,0 +1,224 @@
+        'CNAME' => DNS_CNAME,
+        'AAAA' => DNS_AAAA,
+        'A6' => DNS_A6,
+        'SRV' => DNS_SRV,
+        'NAPTR' => DNS_NAPTR,
+        'TXT' => DNS_TXT,
+        'HINFO' => DNS_HINFO,
+    );
+
+    /**
+     * Configures the mock values for DNS queries.
+     *
+     * @param array $hosts Mocked hosts as keys, arrays of DNS records as returned by dns_get_record() as values.
+     */
+    public static function withMockedHosts(array $hosts)
+    {
+        self::$hosts = $hosts;
+    }
+
+    public static function checkdnsrr($hostname, $type = 'MX')
+    {
+        if (!self::$hosts) {
+            return \checkdnsrr($hostname, $type);
+        }
+        if (isset(self::$hosts[$hostname])) {
+            $type = strtoupper($type);
+
+            foreach (self::$hosts[$hostname] as $record) {
+                if ($record['type'] === $type) {
+                    return true;
+                }
+                if ('ANY' === $type && isset(self::$dnsTypes[$record['type']]) && 'HINFO' !== $record['type']) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    public static function getmxrr($hostname, &$mxhosts, &$weight = null)
+    {
+        if (!self::$hosts) {
+            return \getmxrr($hostname, $mxhosts, $weight);
+        }
+        $mxhosts = $weight = array();
+
+        if (isset(self::$hosts[$hostname])) {
+            foreach (self::$hosts[$hostname] as $record) {
+                if ('MX' === $record['type']) {
+                    $mxhosts[] = $record['host'];
+                    $weight[] = $record['pri'];
+                }
+            }
+        }
+
+        return (bool) $mxhosts;
+    }
+
+    public static function gethostbyaddr($ipAddress)
+    {
+        if (!self::$hosts) {
+            return \gethostbyaddr($ipAddress);
+        }
+        foreach (self::$hosts as $hostname => $records) {
+            foreach ($records as $record) {
+                if ('A' === $record['type'] && $ipAddress === $record['ip']) {
+                    return $hostname;
+                }
+                if ('AAAA' === $record['type'] && $ipAddress === $record['ipv6']) {
+                    return $hostname;
+                }
+            }
+        }
+
+        return $ipAddress;
+    }
+
+    public static function gethostbyname($hostname)
+    {
+        if (!self::$hosts) {
+            return \gethostbyname($hostname);
+        }
+        if (isset(self::$hosts[$hostname])) {
+            foreach (self::$hosts[$hostname] as $record) {
+                if ('A' === $record['type']) {
+                    return $record['ip'];
+                }
+            }
+        }
+
+        return $hostname;
+    }
+
+    public static function gethostbynamel($hostname)
+    {
+        if (!self::$hosts) {
+            return \gethostbynamel($hostname);
+        }
+        $ips = false;
+
+        if (isset(self::$hosts[$hostname])) {
+            $ips = array();
```

Comment:
This could now return an empty array. I guess that is not possible using PHP's built-in function, right? Should we still return `false` at the end if `$ips` is empty?